catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/activity/rtl/MultilingualTest.java

```diff
@@ -0,0 +1,138 @@
+ */
+
+package org.catrobat.catroid.uiespresso.activity.rtl;
+
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.ui.Multilingual;
+import org.catrobat.catroid.uiespresso.util.BaseActivityInstrumentationRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Locale;
+
+import static android.support.test.espresso.Espresso.onData;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.assertThat;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static junit.framework.Assert.assertTrue;
+
+import static org.catrobat.catroid.uiespresso.util.UiTestUtils.getResourcesString;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.hasToString;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.startsWith;
+
+@RunWith(AndroidJUnit4.class)
+public class MultilingualTest {
+	private static final Locale ARABICLOCALE = new Locale("ar");
+	private static final Locale URDULOCALE = new Locale("ur");
+	private static final Locale FARSILOCALE = new Locale("fa");
+	private static final Locale DEUTSCHLOCALE = Locale.GERMAN;
+	@Rule
+	public BaseActivityInstrumentationRule<Multilingual> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(Multilingual.class);
+
+	@Before
+	public void setUp() throws Exception {
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Test
+	public void testChangeLanguageToDeutsch() {
+		onData(hasToString(startsWith(DEUTSCHLOCALE.getDisplayName(DEUTSCHLOCALE)))).perform(click());
+		assertThat("current Locale is not Deutsch", Locale.getDefault().getDisplayLanguage(), is(DEUTSCHLOCALE.getDisplayLanguage()));
+		String buttonProgramsName = getResourcesString(R.string.main_menu_programs);
+		String buttonContinueName = getResourcesString(R.string.main_menu_continue);
+		String buttonNewName = getResourcesString(R.string.main_menu_new);
+		onView(withId(R.id.main_menu_button_continue))
+				.check(matches(withText(containsString(buttonContinueName))));
+		onView(withId(R.id.main_menu_button_new))
+				.check(matches(withText(buttonNewName)));
+		onView(withId(R.id.main_menu_button_programs))
+				.check(matches(withText(buttonProgramsName)));
+	}
+
+	@Test
+	public void testChangeLanguageToArabic() {
+		onData(hasToString(startsWith(ARABICLOCALE.getDisplayName(ARABICLOCALE)))).perform(click());
+		assertThat("current Locale is not Arabic", Locale.getDefault().getDisplayLanguage(), is(ARABICLOCALE.getDisplayLanguage()));
+		assertTrue("current Text direction is not RTL", isRTL(Locale.getDefault()));
```

Comment:
im not a huge fan of assertion messages because most of the times, if the assertion fails, they tell me as much as the the failing assertion does already (in the best case, see my points below).
eg.
assertTrue(isRTL()) 
failing tells me that it seems to not be RTL
while the message ""current Text direction is not RTL"" tells me something that the code already does. it just makes everything a little bit harder to read because it makes everything a bit more cluttered.

someone once said assertion messages are merely comments. in Martin's clean code book theres not a single assertion with an assertion message. this makes sense, considering the comments point, comments lie. code is updated, the comments or in this case the messages often forgotten. and ive seen this a few times already in this project already that the assertion or the values have been adapted for some reason, but the message was forgotten. so the assertion error message actually dumps non-sensical information in the best case, or misleading info in the worst case.
so if theres not a really good reason to have an assertion message, i would always opt for just leaving it out.

regarding my aversion to assertThat.
it has its valid usages, some parts, if used properly it can help readability.
eg.
assertThat( thisString, is("bla") )
is ok ish, but id still prever
assertEquals("bla", thisString) because it makes the point of the assertion, equality, jump at you.

especially when dealing with more complex constructs, eg, the line you had above,
assertThat("current Locale is not Arabic", Locale.getDefault().getDisplayLanguage(), is(ARABICLOCALE.getDisplayLanguage())); 

ok, so lets pretend, this breaks at some point a someone less experienced picks this up as his first bug ticket.
so, not looking at the actual test execution, just having the code speak to itself, its hard to grasp what exactly is checked. since its maybe not clear that Locale.getDefault().getDisplayLanguage() returns a string per se, and keep in mind that there is a lot of hamcrest matchers like "isA" or "isIn" or "isOneOf" just stating a few starting with the same letter... so anyways, my point, the "is" which is actually a shortcut for "is(equalTo(x))", the tipping point of the assertion, is hidden somewhere in this very long line of code.

so just looking at the code:
assertThat("current Locale is not Arabic", Locale.getDefault().getDisplayLanguage(), is(ARABICLOCALE.getDisplayLanguage())); 

vs

assertThat(Locale.getDefault().getDisplayLanguage(), is(ARABICLOCALE.getDisplayLanguage())); 

vs

assertEquals(ARABICLOCALE.getDisplayLanguage(), Locale.getDefault().getDisplayLanguage()); 

or even better (could be work according to https://developer.android.com/reference/java/util/Locale.html if language and country are same)

 assertEquals(ARABICLOCALE, Locale.getDefault())

so to sum it up, a good name for the test, and a properly chosen assertion should make an assertion message unnecessary in 90% of the cases.